### ISSUE TITLE
IngressRoute: TCPProxy Delegation from a root IngressRoute

### ISSUE THREAD

**Issue by glerchundi** [heptio/contour#839]
Trying `tcpproxy` thingy in our deployments I found myself trying to figure out how to make it in our own methodology (which uses delegation for everything so namespace based access control is much easier), I saw in #787 that root delegation is out of scope. Is this something that you would eventually do? I can take some time to sketch something. The `spec` seems to fairly clear to me which would be a mimic of `routes`:

```
[...]
        virtualhost: {
          fqdn: "mydomain.com",
          tls: {
            secretName: "myingress-tls"
          }
        },
        tcpproxy: {
          delegate: {
            name: "myingress",
            namespace: "myingress-system"
          }
        }
      }
[...]
```

WDYT?

**Comment by davecheney** [comment]
@glerchundi this is done, right?

**Comment by davecheney** [comment]
Closed by #851
